packages/collector/test/tracing/sdk/entry_span_context_app.js

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env node
+/*
+ * (c) Copyright IBM Corp. 2022
+ */
+
+'use strict';
+
+const instana = require('../../..')();
+
+const express = require('express');
+const morgan = require('morgan');
+const { delay, getLogger } = require('@instana/core/test/test_util');
+
+const app = express();
+const logPrefix = `SDK entry span context (${process.pid}):\t`;
+const log = getLogger(logPrefix);
+
+const spanName = 'span-name';
+const parentContextAnnotation = 'sdk.custom.tags.parent-context';
+const defaultDelayBetweenIterations = 10;
+const defaultDurationOfSpans = 5;
+
+let delayBetweenIterations = defaultDelayBetweenIterations;
+if (process.env.DELAY) {
+  delayBetweenIterations = parseInt(process.env.DELAY, 10);
+  if (Number.isNaN(delayBetweenIterations)) {
+    delayBetweenIterations = defaultDelayBetweenIterations;
+  }
+}
+
+let durationOfSpans = defaultDurationOfSpans;
+if (process.env.SPAN_DURATION) {
+  durationOfSpans = parseInt(process.env.DELAY, 10);
+  if (Number.isNaN(durationOfSpans)) {
+    durationOfSpans = defaultDelayBetweenIterations;
+  }
+}
+
+if (process.env.WITH_STDOUT) {
+  app.use(morgan(`${logPrefix}:method :url :status`));
+}
+
+app.get('/', (req, res) => {
+  res.sendStatus(200);
+});
+
+async function createSdkSpanRecursiveAsync(iterationsDone, maxIterations) {
+  await instana.sdk.async.startEntrySpan(spanName);
+  annotateWithParentContext();
+  await delay(durationOfSpans);
+  instana.sdk.async.completeEntrySpan();
+
+  iterationsDone++;
+  if (iterationsDone < maxIterations) {
+    await new Promise(resolve => {
+      setTimeout(() => {
+        return createSdkSpanRecursivePromise(iterationsDone, maxIterations).then(resolve);
+      }, delayBetweenIterations);
+    });
+  }
+}
+
+async function createSdkSpanNonRecursiveAsync() {
+  await instana.sdk.async.startEntrySpan(spanName);
+  annotateWithParentContext();
+  await delay(durationOfSpans);
```

Comment:
_question_: I was wondering if we want to add an exit span. This would guarantee that exit spans still keep the reference to it's parent entry span.

Reply:
Various combinations of entry, exit and intermediate spans are already tested in this [test](https://github.com/instana/nodejs/blob/main/packages/collector/test/tracing/sdk/test.js) and [app](https://github.com/instana/nodejs/blob/main/packages/collector/test/tracing/sdk/app.js), I don't think we need to add this aspect here, too -- it would be redundant IMO.

Although that other test does not explicitly handly the `async` api, I'll add that there for completeness.

WDYT?

Reply:
Perfect 👍

Reply:
I'll add those tests as part of a follow up PR.